samples/nrf_desktop/src/modules/ble_adv.c

```diff
@@ -136,25 +151,64 @@ static int ble_adv_start(void)
 
 	if (err) {
 		LOG_ERR("Advertising failed to start (err %d)", err);
-		return err;
+	} else {
+		if (IS_ENABLED(CONFIG_DESKTOP_BLE_FAST_ADV)) {
+			k_delayed_work_submit(&adv_update,
+					      K_SECONDS(CONFIG_DESKTOP_BLE_FAST_ADV_TIMEOUT));
+			state = STATE_ACTIVE_FAST;
+		} else {
+			state = STATE_ACTIVE_SLOW;
+		}
+
+		LOG_INF("Advertising started");
 	}
 
-	LOG_INF("Advertising started");
+	return err;
+}
+
+static void sp_grace_period_fn(struct k_work *work)
+{
+	int err = ble_adv_stop();
 
-	if (IS_ENABLED(CONFIG_DESKTOP_BLE_SWIFT_PAIR)) {
-		k_delayed_work_submit(&adv_update,
-			K_SECONDS(CONFIG_DESKTOP_BLE_FAST_ADV_TIMEOUT));
+	if (err) {
+		module_set_state(MODULE_STATE_ERROR);
+	} else {
+		module_set_state(MODULE_STATE_OFF);
 	}
+}
+
+static int remove_vendor_section(void)
+{
+	int err = bt_le_adv_update_data(ad, (ARRAY_SIZE(ad) - SWIFT_PAIR_SECTION_SIZE),
+					sd, ARRAY_SIZE(sd));
+
+	if (err) {
+		LOG_ERR("Cannot modify advertising data (err %d)", err);
+	} else {
+		LOG_INF("Vendor section removed");
+
+		if (IS_ENABLED(CONFIG_DESKTOP_BLE_FAST_ADV)) {
```

Comment:
This if statement seems redundant.

This function can only be called if CONFIG_DESKTOP_BLE_SWIFT_PAIR is enabled
&
CONFIG_DESKTOP_BLE_SWIFT_PAIR depends on CONFIG_DESKTOP_BLE_FAST_ADV

Reply:
this is true; I have added it for symmetry as the dependency is hidden in the Kconfig and may not be obvious; If you want I can remove it so please tell

Reply:
you can leave it like it is now

Reply:
thx